jsonschema/tests/test_jsonschema_test_suite.py

```diff
@@ -401,15 +401,7 @@ def leap_second(test):
     skip=lambda test: (
         narrow_unicode_build(test)
         or skip(
-            message="dynamicRef support isn't working yet.",
-            subject="dynamicRef",
-        )(test)
-        or skip(
-            message="These tests depends on dynamicRef working.",
-            subject="defs",
-        )(test)
-        or skip(
-            message="These tests depends on dynamicRef working.",
+            message="These tests require an extension or the url resolver.",
```

Comment:
Thanks! This looks promising. Can you elaborate on what you mean by an extension of the URL resolver? Looking at this test it doesn't look dynamicRef related (so I must have mis-labelled the message), but I'm not sure I see why this wouldn't work as-is with the location independent ref support you helped add.

Reply:
The `resolve_fragment` method located in `RefResolver` is resolving the anchor by taking the first subschema that matches the anchor name in the current schema. 

```python
for keyword in ["$anchor", "$dynamicAnchor"]:
    for subschema in self._finditem(document, keyword):
        if fragment == subschema[keyword]:
            return subschema
```

This work fine as long as the anchor names are not repeated in other scopes, this is why the test fails and resolves to:

```json
{
    "$id": "child2",
    "$anchor": "my_anchor",
    "type": "number"
}
```

We may need to extend `_finditem` to to not leave the scope while searching for the anchor or take all findings and evaluate if they not break out of the current scope before returning the result.